src/commands/handler.js

```diff
@@ -41,13 +41,21 @@ module.exports = class IrcCommandHandler extends EventEmitter {
         if (batch_id) {
             const cache_key = 'batch.' + batch_id;
             if (this.hasCache(cache_key)) {
-                const cache = this.cache(cache_key);
+                let cache = this.cache(cache_key);
                 cache.commands.push(irc_command);
+                while (cache?.tags?.batch) {
```

Comment:
Call me old fashioned, but I like my caches to not change while I'm reading them.

Are you sure this is what you want to do ?